### ISSUE TITLE
For swift generation, allow user to specify access control

### ISSUE THREAD

**Issue by adysart** [apollographql/apollo-codegen#278]
Currently, in generated swift code all types are created as `public`. This makes it difficult to encapsulate these generated types within a module. It would be nice if when invoking apollo-codegen you could specify an access control level:

    apollo-codegen generate foo.graphql --schema schema.json --output API.swift -accesscontrol internal

**Comment by sirghi** [comment]
Agree. We're building a framework and we want to abstract away the Apollo generated types. This is a very desired feature

**Comment by sirghi** [comment]
@jbaxleyiii  just closing it, without any comment/feedback?

**Comment by stevenp** [comment]
This is still a critical problem. Can it be reopened? If not, I'll go ahead and open a new issue.

**Comment by bentu-noodoe** [comment]
Same. We're packaging a framework to our client, and we don't want to expose our implementation details. 
It'd be great to have official support here.

**Comment by lucastimeless** [comment]
> This is still a critical problem. Can it be reopened? If not, I'll go ahead and open a new issue.

This is a problem for me as well, was a new ticket opened?

**Comment by designatednerd** [comment]
@lucastimeless This has been moved over to the iOS repo, I've already got this working in the parts of the iOS codegen I've moved to swift (which I will grant you is not much right now)
